File: scala/armstrong-numbers/src/main/scala/ArmstrongNumbers.scala
import scala.math.pow

object ArmstrongNumbers {
  def isArmstrongNumber(number: Int) : Boolean = {
    val numAsString = number.toString
    val length = numAsString.length
    numAsString
          .map(digit => pow(digit.asDigit, length).asInstanceOf[Int])
          .sum
          .equals(number)
  }
}

File: lua/armstrong-numbers/armstrong-numbers.lua
local ArmstrongNumbers = {}

function ArmstrongNumbers.is_armstrong_number(number)
	local total, digits = 0, tostring(number)
	for c in string.gmatch(digits, ".") do
		total = total + tonumber(c) ^ #digits
	end
	return total == number
end

return ArmstrongNumbers
